execution_engine/src/storage/error/lmdb.rs

```diff
@@ -46,6 +51,22 @@ impl From<in_memory::Error> for Error {
         match error {
             in_memory::Error::BytesRepr(error) => Error::BytesRepr(error),
             in_memory::Error::Poison => Error::Poison,
+            in_memory::Error::MerkleConstruction(error) => Error::MerkleConstruction(error),
+        }
+    }
+}
+
+impl From<MerkleConstructionError> for Error {
+    fn from(error: MerkleConstructionError) -> Self {
+        Error::MerkleConstruction(error)
+    }
+}
```

Comment:
```suggestion
```

(Just use `#[from]` in the proc macro)

Reply:
I did initially, but then the compiler complained about the `"{0}"` in the attribute, that some traits aren't implemented. So I just followed the example of the BytesRepr variant and it works this way.

Reply:
Maybe you could elide `"{0}"` and use `transparent` instead... what error are you getting exactly?

Reply:
Huh, interesting. It worked this time when I applied your suggestion, but when I tried to do the same for the `BytesRepr` variant for consistency, I got this:
```
error[E0599]: the method `as_dyn_error` exists for reference `&casper_types::bytesrepr::Error`, but its trait bounds were not satisfied
   --> execution_engine/src/storage/error/lmdb.rs:19:13
    |
19  |     #[error("{0}")]
    |             ^^^^^ method cannot be called on `&casper_types::bytesrepr::Error` due to unsatisfied trait bounds
    |
   ::: /home/bartek/CasperLabs/casper-node/types/src/bytesrepr.rs:110:1
    |
110 | pub enum Error {
    | --------------
    | |
    | doesn't satisfy `_: thiserror::private::AsDynError`
    | doesn't satisfy `casper_types::bytesrepr::Error: std::error::Error`
    |
    = note: the following trait bounds were not satisfied:
            `casper_types::bytesrepr::Error: std::error::Error`
            which is required by `casper_types::bytesrepr::Error: thiserror::private::AsDynError`
            `&casper_types::bytesrepr::Error: std::error::Error`
            which is required by `&casper_types::bytesrepr::Error: thiserror::private::AsDynError`
```
I guess the issue is that `bytesrepr::Error` doesn't implement `Error`?